eclair-core/src/main/scala/fr/acinq/eclair/channel/ChannelEvents.scala

```diff
@@ -32,7 +32,7 @@ trait ChannelEvent
 
 case class ChannelCreated(channel: ActorRef, peer: ActorRef, remoteNodeId: PublicKey, isFunder: Boolean, temporaryChannelId: ByteVector32, initialFeeratePerKw: FeeratePerKw, fundingTxFeeratePerKw: Option[FeeratePerKw]) extends ChannelEvent
 
-case class ChannelRestored(channel: ActorRef, channelId: ByteVector32, peer: ActorRef, remoteNodeId: PublicKey, isFunder: Boolean, currentData: AbstractCommitments) extends ChannelEvent
+case class ChannelRestored(channel: ActorRef, channelId: ByteVector32, peer: ActorRef, remoteNodeId: PublicKey, isFunder: Boolean, currentData: HasAbstractCommitments) extends ChannelEvent
```

Comment:
Ins't it `ChannelStateChanged` that you should update?

Reply:
I was looking where this trait would be appropriate, applying it here saves a bit of code here: https://github.com/ACINQ/eclair/pull/1580/files#diff-9242cfc225576251a04aa532dfe001172c5b6614dd9411c7c965d59457130f23L196

but indeed if `HasAbstractCommitments` is to be thrown away this event will be just fine without it.

Reply:
As to `ChannelStateChanged`, it needs not be updated if `HasAbstractCommitments` is to stay since it uses `currentData: Data` and `ChannelStateChanged` extends from `Data` so can be published from base channels as well as plugins.